activerecord/lib/active_record/connection_adapters/mysql2/database_statements.rb

```diff
@@ -48,7 +48,24 @@ def exec_delete(sql, name = nil, binds = []) # :nodoc:
             exec_stmt_and_free(sql, name, binds) { |stmt| stmt.affected_rows }
           end
         end
-        alias :exec_update :exec_delete
+
+        def exec_update(sql, name = nil, binds = [], _returning = nil) # :nodoc:
+          affected_rows = if without_prepared_statement?(binds)
+            with_raw_connection do |conn|
+              @affected_rows_before_warnings = nil
+              execute_and_free(sql, name) do
+                @affected_rows_before_warnings || conn.affected_rows
+              end
+            end
+          else
+            exec_stmt_and_free(sql, name, binds) do |stmt|
+              stmt.affected_rows
+            end
+          end
```

Comment:
This is essentially a copy of `exec_delete`, right? 
So technically the method could be written as:

```suggestion
     exec_delete(...)
     build_result(columns: [], rows: [], affected_rows: affected_rows)
```

Though it does look weird so I think instead the logic from `exec_delete` should be extracted into a new method like:

```ruby
def _exec_statement(sql, name = nil, binds = [], _returning = nil)
   <same body as current version of `exec_delete`>
end

alias :exec_delete :_exec_statement

def exec_update(sql, name = nil, binds = [], _returning = nil) # :nodoc:
    affected_rows = exec_statement(...)
    build_result(columns: [], rows: [], affected_rows: affected_rows)
end
```

Reply:
Yes, it is a copy. I wasn't sure how closely the Rails codebase followed the rule of three. So, I elected to open the PR without an abstraction and see if there was feedback.

I like the suggestion as written and will implement it as such.